centrallix/osdrivers/objdrv_http.c

```diff
@@ -1712,10 +1719,22 @@ http_internal_GetPageStream(pHttpData inf)
 		inf->ContentType = nmSysStrdup(hdr_val);
 		if (strchr(inf->ContentType, ';'))
 		    *(strchr(inf->ContentType, ';')) = '\0';
+		/* add charset if present */
+		char * charsetStart = NULL;
+		if((charsetStart = strstr(hdr_val, "charset=")) != NULL && !inf->OverrideContentCharset)
+		    {
+		    inf->ContentCharset = nmSysStrdup(charsetStart+8);
+		    if (strchr(inf->ContentCharset, ';'))
+			*(strchr(inf->ContentCharset, ';')) = '\0';
+		    }
+		else if(inf->ExpectedContentCharset)
+		    inf->ContentCharset = nmSysStrdup(inf->ExpectedContentCharset);
```

Comment:
Is this the line you were referring to? The way the line wraps on my screen makes it harder to notice it is actually inside the NULL check on the line 1730